styx-common/src/main/java/com/spotify/styx/api/Middlewares.java

```diff
@@ -79,39 +90,94 @@ private Middlewares() {
     });
   }
 
-  public static Middleware<AsyncHandler<? extends Response<?>>,
-      AsyncHandler<? extends Response<ByteString>>> clientValidator(
+  public static <T> Middleware<AsyncHandler<Response<T>>, AsyncHandler<Response<T>>> clientValidator(
       Supplier<Optional<List<String>>> supplier) {
     return innerHandler -> requestContext -> {
       if (requestContext.request().header("User-Agent")
           .map(header -> supplier.get().orElse(ImmutableList.of()).contains(header))
           .orElse(false)) {
         // TODO: fire some stats
-        return CompletableFuture
-            .completedFuture(Response.forStatus(Status.NOT_ACCEPTABLE.withReasonPhrase(
+        return
+            completedFuture(Response.forStatus(Status.NOT_ACCEPTABLE.withReasonPhrase(
                 "blacklisted client version, please upgrade")));
       } else {
-        // noinspection unchecked
-        return (CompletionStage<Response<ByteString>>) innerHandler.invoke(requestContext);
+        return innerHandler.invoke(requestContext);
       }
     };
   }
 
-  public static Middleware<AsyncHandler<? extends Response<?>>,
-      AsyncHandler<? extends Response<ByteString>>> auditLogging() {
+  private static final GoogleIdTokenVerifier GOOGLE_ID_TOKEN_VERIFIER;
+
+  static {
+    final NetHttpTransport transport;
+    try {
+      transport = GoogleNetHttpTransport.newTrustedTransport();
+    } catch (GeneralSecurityException | IOException e) {
+      throw new RuntimeException(e);
+    }
+    GOOGLE_ID_TOKEN_VERIFIER = new GoogleIdTokenVerifier
```

Comment:
I believe so. I've never seen code that deals with transport level connections on this layer of their client libs. Seems it's all handled within the `transport` implementation.